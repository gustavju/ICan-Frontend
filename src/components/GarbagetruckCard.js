import React from 'react';

const GarbagetruckCard = ({ selectGarbagetruck, garbagetruck }) => (
    <div onClick={(e) => selectGarbagetruck(e, garbagetruck.garbageTruckId)} className="garbagetruck-card">
        <div className="garbagetruck-card__id">{garbagetruck.garbageTruckId}</div>
        <div className="garbagetruck-card__text"><span>Capacity:</span>{garbagetruck.capacity}</div>
        <div className="garbagetruck-card__text"><span>Location:</span>{garbagetruck.location.longitude}, {garbagetruck.location.latitude}</div>
    </div>
);

export default GarbagetruckCard;