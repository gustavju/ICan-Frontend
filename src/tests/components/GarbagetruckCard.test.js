import React from 'react';
import { shallow } from 'enzyme';
import GarbagetruckCard from '../../components/GarbagetruckCard';
import garbagetrucks from '../fixtures/garbagetrucks';

test('should render the GarbagetruckCard', () => {
    const wrapper = shallow(<GarbagetruckCard garbagetruck={garbagetrucks[0]} />);
    expect(wrapper).toMatchSnapshot();
});