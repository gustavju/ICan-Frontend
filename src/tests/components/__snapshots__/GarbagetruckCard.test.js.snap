// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the GarbagetruckCard 1`] = `
<div
  className="garbagetruck-card"
  onClick={[Function]}
>
  <div
    className="garbagetruck-card__id"
  >
    3b2617d3-d9fc-42cb-b3ee-ab3f074598df
  </div>
  <div
    className="garbagetruck-card__text"
  >
    <span>
      Capacity
    </span>
    0.0
  </div>
  <div
    className="garbagetruck-card__text"
  >
    <span>
      Location
    </span>
    17.944543
    , 
    59.407136
  </div>
</div>
`;
